samples/socket-api/README.md

```diff
@@ -0,0 +1,57 @@
+"socket-api" sample introduction
+================================
+
+This sample demonstrates how to use WAMR socket-api to develop wasm network applications.
+Two wasm applications are provided: tcp-server and tcp-client, and this sample demonstrates
+how they communicate with each other.
+
+## Preparation
+
+Please install WASI SDK, download the [wasi-sdk release](https://github.com/CraneStation/wasi-sdk/releases) and extract the archive to default path `/opt/wasi-sdk`.
+And install wabt, download the [wabt release](https://github.com/WebAssembly/wabt/releases) and extract the archive to default path `/opt/wabt`
+
+## Build the sample
+
+```bash
+mkdir build
+cd build
+cmake ..
+make
+```
+
+The file `tcp_server.wasm`, `tcp_client.wasm` and `iwasm` will be created.
+And also file `tcp_server` and `tcp_client` of native version are created.
+
+Note that iwasm is built with libc-wasi and lib-pthread enabled.
+
+## Run workload
+
+Start the tcp server, which opens port 1234 and waits for clients to connect.
+```bash
+cd build
+./iwasm --addr-pool=127.0.0.1/15 tcp_server.wasm
```

Comment:
The server should use 0.0.0.0 to allow it to bind to any address. Partially because the code is use 0.0.0.0.